### DISCUSSION TITLE
In the UI, now that we have warehouse object in the config i think it should also be shown in the UI

### DISCUSSION THREAD

**Discussion by tal-hason** [akuity/kargo#1156] [Ideas]
since 0.2.1 we have warehouse objects that combine a specific set of subscriptions(image, git & helm), but in the UI we only get the same old visual of the separated items without the dedicated view of the relations of each item to its warehouse relation.

I think those combine items from a single warehouse, which creates a single freight for any change in one of the items.

for example:

the current view:

![image](https://github.com/akuity/kargo/assets/97454240/78e067fc-2eb2-4c89-99f7-757ee3f84887)

My vision of a warehouse view:
![image](https://github.com/akuity/kargo/assets/97454240/85f906b3-d66a-485b-850a-d503948daab6)

let me know what you think

**Comment by krancour** [comment]
@rbreeze and I had discussed this @tal-hason and I think we agreed we wanted something like this, but iirc, there were some technical barriers to resolve first. @rbreeze, do I recall correctly?

**Reply by tal-hason** [reply]
@krancour, after a little more thought about it, maybe even connect the "freight-line" bar to a warehouse, so each time the user selects a warehouse in the UI the freight-line bar will only show the relevant freight for this warehouse.

**Reply by krancour** [reply]
@tal-hason ya... especially as the idea of multiple pipelines in a single project begins to be explored more earnestly, we'll need to figure out how to display multiple freightlines -- one per warehouse. With limited real estate, I don't think we've quite figured out how yet, but it's on our radar.

**Reply by tal-hason** [reply]
check discord i sent you a message

**Reply by tal-hason** [reply]
@krancour after our quick chat in discord, did you have any insights about my idea?

**Reply by rbreeze** [reply]
@tal-hason I like the idea of filtering the Freightline by warehouse. I'll open an issue for this to track it if it's okay with you.

@krancour You're right, the technical issue is that it's difficult to scale the DAG nodes dynamically. So we'd have to do some trickery to calculate the right size DAG node based on the amount of subscriptions in the warehouse. It's doable, but more effort than the current implementation

**Reply by rbreeze** [reply]
https://github.com/akuity/kargo/issues/1208

**Comment by krancour** [comment]
Fixed long ago.
